overlord/fdestate/backend/reseal.go

```diff
@@ -236,53 +235,42 @@ func doReseal(manager FDEStateManager, method device.SealingMethod, rootdir stri
 		}
 	}
 
-	switch method {
-	case device.SealingMethodFDESetupHook:
-		// Note this is the fallback key path for old installations. The file might not exist.
-		primaryKeyFile := filepath.Join(boot.InstallHostFDESaveDir, "aux-key")
-		// All devices should have the same primary key
-		primaryKeyGetter := func() ([]byte, error) {
-			return secbootGetPrimaryKey(devices, primaryKeyFile)
-		}
-		for _, key := range keys {
-			if err := secbootResealKeysWithFDESetupHook([]secboot.KeyDataLocation{key.location}, primaryKeyGetter, key.params.Models, key.params.BootModes); err != nil {
-				return err
-			}
+	saveFDEDir := dirs.SnapFDEDirUnderSave(dirs.SnapSaveDirUnder(rootdir))
+	fallbackPrimaryKeyFiles := []string{
+		filepath.Join(saveFDEDir, "aux-key"),
+		filepath.Join(saveFDEDir, "tpm-policy-auth-key"),
+	}
+
+	var allResealedKeys secboot.UpdatedKeys
+	for _, key := range keys {
+		params := &secboot.ResealKeyParams{
+			PrimaryKeyDevices:       devices,
```

Comment:
we might want a comment that explains why we use all devices here and not just the device associated with the key.location?